d2g.go

```diff
@@ -43,6 +43,14 @@ func Convert(dwPayload *dockerworker.DockerWorkerPayload) (gwPayload *genericwor
 	if err != nil {
 		return
 	}
+	switch dwImage.(type) {
+	case *IndexedDockerImage:
+		// we want to be sure that TaskclusterProxy
+		// is enabled for IndexedDockerImages
+		// during the remainder of this translation
+		// it's used to access the index service API
+		dwPayload.Features.TaskclusterProxy = true
```

Comment:
Can we use something like `taskclusterProxyRequired` instead, so that the docker-worker payload isn't mutated? Or maybe we can just set `gwPayload.Features.TaskclusterProxy` to `true`, and then later, where we have e.g.

```go
	if dwPayload.Features.TaskclusterProxy {
		dwManagedEnvVars = append(dwManagedEnvVars, "TASKCLUSTER_PROXY_URL")
	}
```

instead we would check `gwPayload.Features.TaskclusterProxy`?

... We would also have to change the setting of the property to something like

```
gwPayload.Features.TaskclusterProxy = gwPayload.Features.TaskclusterProxy || dwPayload.Features.TaskclusterProxy
```

in case it had already been set...

Reply:
Yeah I'll apply these suggestions, thanks Pete! Gettin a _little bit_ hacky lol

Reply:
Addressed in https://github.com/taskcluster/d2g/pull/18/commits/2c86eaaa6d87eaa44c079849d73ff5e0cee3af8d

Reply:
Sorry, just realised it is correct to set `gwPayload.Features.TaskclusterProxy` to `true`, but I was wrong about:

```
	if dwPayload.Features.TaskclusterProxy {
		dwManagedEnvVars = append(dwManagedEnvVars, "TASKCLUSTER_PROXY_URL")
	}
```

That should have stayed as it was - my bad! The reason is, the podman run command shouldn't have access to the taskcluster proxy, if it didn't under docker-worker. The access to taskcluster proxy is only needed for the `curl` command that downloads the container image from the Index service...

Reply:
Addressed in https://github.com/taskcluster/d2g/pull/18/commits/50f96ad3311d277d557191f8bdff4cd8b787d2de and also simplified the inputs to `podmanEnvMappings()`.